### PR TITLE
Transitioning from List to Map based implementation in kafka-topic-creation hook 

### PR DESCRIPTION
**Pull request by japoorv**
Currently we have a list implementation of the the various kafka-topics which needs to be generated. This makes changing certain values of the topic configurations difficult since list are not merged in Helm. So migration of the list to map should be done as maps support merging.

This is the proposed format for the way topic configurations be declared in their respective producer helm charts. 
```
kafka-topic-creator:
  kafka:
    topics:
      backend-entity-view-events:
        replicationFactor: 3
        partitions: 8
        configs:
          retention.bytes: 1073741824
          retention.ms: 21600000
      raw-trace-view-events:
        replicationFactor: 3
        partitions: 8
        configs:
          retention.bytes: 1073741824
          retention.ms: 21600000
      raw-service-view-events:
        replicationFactor: 3
        partitions: 8
        configs:
          retention.bytes: 1073741824
          retention.ms: 21600000
      service-call-view-events:
        replicationFactor: 3
        partitions: 8
        configs:
          retention.bytes: 1073741824
          retention.ms: 21600000
      span-event-view:
        replicationFactor: 3
        partitions: 8
        configs:
          retention.bytes: 1073741824
          retention.ms: 21600000
      log-event-view:
        replicationFactor: 3
        partitions: 8
        configs:
          retention.bytes: 1073741824
          retention.ms: 21600000
```

### CODE REVIEW

File: helm/templates/kafka-topic-creation-hook.yaml [hunk 1]
```diff
@@ -37,25 +37,26 @@ spec:
             - "/bin/bash"
             - "-cex"
             - |
-            {{- range $topic := .Values.kafka.topics }}
+            {{- range $topicName, $configs := $.Values.kafka.topics }}
```
Comment (laxmanchekka) [inline helm/templates/kafka-topic-creation-hook.yaml]:
```suggestion
            {{- range $topicName, $topicConfigs := $.Values.kafka.topics }}
```

File: helm/templates/kafka-topic-creation-hook.yaml [hunk 2]
```diff
@@ -37,25 +37,26 @@ spec:
...
               --zookeeper {{ $.Values.zookeeper.address | quote }} \
               --add-config \
-              {{ range $i, $c := $topic.configs }}{{ if $i }}{{ printf "," }}{{ end }}{{ $c | quote }}{{ end }} \
+              {{ range $i, $c := $topicConfigs.configs }}{{ if $comma }}{{ printf "," }}{{ end }}"{{ $i }}={{ int $c }}"{{- $comma = true -}}{{ end }} \
               --entity-type topics \
-              --entity-name {{ $topic.name | quote }}
-            {{- end }}
-            {{- end }}
+              --entity-name {{ $topicName | quote }}
+          {{- end }}
```
Comment (laxmanchekka) [inline helm/templates/kafka-topic-creation-hook.yaml:58]: nit: check formatting/indentation

Comment (japoorv) [reply helm/templates/kafka-topic-creation-hook.yaml]: Fixed

**Review by laxmanchekka — DISMISSED** [review]

File: helm/Chart.yaml [hunk 3]
```diff
@@ -14,7 +14,7 @@ type: application
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 0.1.0
+version: 0.1.1
```
[the remark this answers is not in the document]
Comment (avinashkolluru) [reply helm/Chart.yaml:17]: Yes. Please revert this.

[the remark this answers is not in the document]
Comment (japoorv) [reply helm/Chart.yaml:17]: Okay. 
But as per the comment shouldn't we upgrade this version ? 
Additionally if we are not changing the version, how will other topic-creator repos (which are still using previous list based approach) be aware of not pulling this ?

[the remark this answers is not in the document]
Comment (ravisingal) [reply helm/Chart.yaml]: This version is updated by CI pipeline. no need to change it here.

File: helm/templates/kafka-topic-creation-hook.yaml [hunk 4]
```diff
@@ -37,23 +37,24 @@ spec:
...
-            {{- if $topic.configs }}
+              --replication-factor {{ $topicConfigs.replicationFactor | quote }} \
+              --partitions {{ $topicConfigs.partitions | quote }}
+            {{- if $topicConfigs.configs }}
+            {{- $comma := false }}
               /opt/kafka/bin/kafka-configs.sh --alter \
               --zookeeper {{ $.Values.zookeeper.address | quote }} \
               --add-config \
-              {{ range $i, $c := $topic.configs }}{{ if $i }}{{ printf "," }}{{ end }}{{ $c | quote }}{{ end }} \
+              {{ range $i, $c := $topicConfigs.configs }}{{ if $comma }}{{ printf "," }}{{ end }}"{{ $i }}={{ int $c }}"{{- $comma = true -}}{{ end }} \
```
Comment (ravisingal) [inline helm/templates/kafka-topic-creation-hook.yaml:55]: you can't update the value of variable `comma`

Comment (ravisingal) [reply helm/templates/kafka-topic-creation-hook.yaml:55]: helm does not allow it.

Comment (japoorv) [reply helm/templates/kafka-topic-creation-hook.yaml:55]: I actually checked it on one stackoverflow posts [(post)](https://stackoverflow.com/questions/64272789/why-doesnt-this-change-the-value-of-the-variable-in-the-range-loop-in-helm) , they say it is. 
Additionally I am getting the right rendering by executing `helm template helm/.`
Are we using older helm version ?

**Review by laxmanchekka — APPROVED** [review]

**Pull request merged**